### PR TITLE
Convert the repository to the new collection - bump version to 2.0.0

### PR DESCRIPTION
**Pull request by aalaesar**
## convert the role into an ansible collection
- moves the role files to the proper folder and adjust CI accordingly
- merge run_occ module developed on the nextcloud repository

### CODE REVIEW

**Comment by aalaesar** [comment]
LGTM
@staticdev @wiktor2200  could you review ?

**Comment by staticdev** [comment]
@aalaesar I started reviewing and have a question.. Why did you change the License type? As far as I see it is moving to a more restrictive than we have today. AGPL License can many times repel good collaborator from open-source community.

File: galaxy.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+namespace: aalaesar
+
+# The name of the collection. Has the same character restrictions as 'namespace'
+name: nextcloud
+
+# The version of the collection. Must be compatible with semantic versioning
+version: 2.0.0
+
+# The path to the Markdown (.md) readme file. This path is relative to the root of the collection
+readme: README.md
```
Comment (staticdev) [inline galaxy.yml:15]: Will this work since readme.md is being deleted from the root in this PR? Is it maybe missing a readme for the collection?

Comment (aalaesar) [reply galaxy.yml:15]: The role's readme have been moved along it and  I created a new one for describing the collection. You should see a large diff for it

**Comment by aalaesar** [comment]
> @aalaesar I started reviewing and have a question.. Why did you change the License type? As far as I see it is moving to a more restrictive than we have today. AGPL License can many times repel good collaborators from open-source community.

Hmm.. I'm not in the known  of OSS licensing  debate (if any). 🤔 
I looked at licenses in some large collections, nextcloud and other projects, compared github's summaries, and I understood  than AGPL was a pretty permissive License,  widely used.
Nothing's engraved in stone yet, so could you kindly enlight me on the community  perception  about agpl and how it changes contributing with this repo compared to the previous  one ?

**Comment by staticdev** [comment]
> > @aalaesar I started reviewing and have a question.. Why did you change the License type? As far as I see it is moving to a more restrictive than we have today. AGPL License can many times repel good collaborators from open-source community.
> 
> Hmm.. I'm not in the known of OSS licensing debate (if any). thinking I looked at licenses in some large collections, nextcloud and other projects, compared github's summaries, and I understood than AGPL was a pretty permissive License, widely used. Nothing's engraved in stone yet, so @staticdev could you kindly enlight me on the community perception about agpl and how it changes contributing with this repo compared to the previous one ?

This is a really complex issue even for lawyers, I think it is a bit waste of time to enter this rabbit role. I worked before for government and today with a big regulated company, and they all have policies of what I am allowed to contribute or use in the internet (even in my private time).

Having that said, my only recommendation would be keep the current one unless you have a very good reason to change. Projects with history of license type change usually are scary in a sense the show instability and it could become proprietary anytime.

**Comment by wiktor2200** [comment]
I agree that licensing may be a problem for some companies or individuals. Maybe not AGPL, but in general it's common problem across products. Some quite loud license changes were Mongo and Elastic/Kibana in the past.
I've taken a look on ansible code, **except python scripts** and it's LGTM.
@staticdev do you have powers to review and validate Python code?
BTW, when we are introducing python code to repository, shouldn't we introduce some CI for that (lint, checks, tests)? Molecule doesn't verify it AFAIK.

**Comment by staticdev** [comment]
Yes I can review python, just the PR is a bit too big and I lack some explanation what else is being made besides transforming the role into a collection. 

@aalaesar would maybe be possible that we split it? Would it make sense to introduce the python modules in a later stage maybe? Or other way.. not sure what would be the most practical strategy here.

**Comment by aalaesar** [comment]
Hello.
I initialy wanted to merge this repo with the nextcloud repo I had started to work on last year.
This previous repository was intended to be the next collection, but as @staticdev suggested, keeping this repo is a good idea to keep stats and visibility.
hence the change of licensing (that was initialy from the other repo) and the python code containing the very first nextcloud ansible module ( 😄 ).
@staticdev Yes I did the two steps in a row that's why the PR is large, but I understand the issue.
I'll rework this that way:

- I'll close this pr
- create a 2.0-dev branch
- create a branch and PR into 2.0-dev to move the role into the proper folder, cherry pick the CI fixes 
- create a branch and PR into 2.0-dev to merge the old collection repo, with proper license.(I really want to ship  the module for the collection first release). Here we will be able to add some _basic_ CI to test the module. if necessary
- merge 2.0-dev to main
- release 2.0
- transfert to nextcloud org

**Comment by aalaesar** [comment]
PR is too much confusing: the PR will be reworked

**Comment by staticdev** [comment]
Thanks @aalaesar, let me know also if I can help you with that.

**Pull request closed without merging**